tools/stress-cluster/services/Stress.Watcher/tests/JobEventHandlerTest.cs

```diff
@@ -0,0 +1,209 @@
+                CreateContainer("init-azure-deployer", env)
+            };
+            handler.ShouldDeleteResources(job).Should().BeTrue();
+
+            env = new List<V1EnvVar> {
+                new V1EnvVar("RESOURCE_GROUP_NAME")
+            };
+            job.Spec.Template.Spec.InitContainers = new List<V1Container>()
+            {
+                CreateContainer("other-init-container"),
+                CreateContainer("init-azure-deployer", env)
+            };
+            handler.ShouldDeleteResources(job).Should().BeTrue();
+
+            env = new List<V1EnvVar> {
+                new V1EnvVar("RESOURCE_GROUP_NAME", "testrg")
+            };
+            job.Spec.Template.Spec.InitContainers = new List<V1Container>()
+            {
+                CreateContainer("other-init-container"),
+                CreateContainer("init-azure-deployer", env)
+            };
+            handler.ShouldDeleteResources(job).Should().BeTrue();
+        }
+
+
+        [Fact]
+        public void TestShouldDeleteResourcesWithCondition()
+        {
+            var handler = new JobEventHandler(null, null, null);
+            var job = CreateJob("testns");
+
+            handler.ShouldDeleteResources(job).Should().BeFalse();
+            var env = new List<V1EnvVar> { new V1EnvVar("RESOURCE_GROUP_NAME", "testrg") };
+            job.Spec.Template.Spec.InitContainers = new List<V1Container>() { CreateContainer("init-azure-deployer", env) };
+
+            job.Status.Conditions = new List<V1JobCondition> { };
+            handler.ShouldDeleteResources(job).Should().BeFalse();
+            job.Status.Conditions = new List<V1JobCondition> { new V1JobCondition("False", "Failed") };
+            handler.ShouldDeleteResources(job).Should().BeFalse();
+            job.Status.Conditions = new List<V1JobCondition> { new V1JobCondition("Unknown", "") };
+            handler.ShouldDeleteResources(job).Should().BeFalse();
+            job.Status.Conditions = new List<V1JobCondition> { new V1JobCondition("False", "Complete") };
+            handler.ShouldDeleteResources(job).Should().BeFalse();
+            job.Status.Conditions = new List<V1JobCondition> { new V1JobCondition("True", "Complete") };
+            handler.ShouldDeleteResources(job).Should().BeTrue();
+        }
+
+        [Fact]
+        public void TestShouldDeleteResourcesWithLabel()
+        {
+            var handler = new JobEventHandler(null, null, null);
+            var job = CreateJob("testns");
+
+            handler.ShouldDeleteResources(job).Should().BeFalse();
+
+            job.Status.Conditions = new List<V1JobCondition> { new V1JobCondition("True", "Failed") };
+            job.Metadata.Labels.Add("Skip.RemoveTestResources", "true");
+            handler.ShouldDeleteResources(job).Should().BeFalse();
+            job.Metadata.Labels.Remove("Skip.RemoveTestResources");
+            job.Spec.Template.Metadata.Labels.Add("Skip.RemoveTestResources", "true");
```

Comment:
Is there a reason why we're allowing Skip.RemoveTestResources @ different places? should/can this be standardized?

Reply:
I want to enable people to put it in their template, and we have two different addons wrappers: one that lets you define from the pod scope (the most common) and one that lets you define from the job scope (e.g. for the net tests). I didn't want to require people use one or the other, so I figured it would be better to support both.

Reply:
Fair enough..